.github/workflows/test.yml

```diff
@@ -54,6 +54,16 @@ jobs:
     name: Build-master
     runs-on: ubuntu-latest
     steps:
+    # Select which dockerfile to use.
+    - name: Use upstream OVN main branch for scheduled jobs
+      if: github.event_name == 'schedule'
+      run: |
+        echo "OVN_DOCKERFILE=Dockerfile.fedora.dev" >> $GITHUB_ENV
+
+    - name: Use Fedora OVN rpms for user triggered jobs
+      if: github.event_name != 'schedule'
+      run: |
+        echo "OVN_DOCKERFILE=Dockerfile.fedora" >> $GITHUB_ENV
```

Comment:
I am completely confused because I think this contradicts what you have commented before:

> The idea was to:
>run PR jobs with ovn-org/ovn main.
>run scheduled jobs with fedora RPMs:`

Did we reverse the decision and I missed that?

Also I thought that we would be using RPMs for ovn-org/ovn main built every 12 hours but not sure now.

Reply:
We had an offline discussion a few weeks ago and we had said we'd start with scheduled jobs with ovn-org/ovn main and keep PR jobs with fedora rpms.  We'd re-evaluate in a while if the scheduled ones turn out to be stable.

I did try to use RPMs but the problem is upstream OVS rpms are built on fedora:37, OVN rpms are built on fedora:38 and the ovn-kubernetes dockerfile uses fedora:37.  The only reasonable choice was then to rebuild the rpms for whatever image the ovn-kubernetes dockerfile was using.  But in that case we might as well just build from source.  It takes the same amount of time and it's less cumbersome (no need to download artifacts from different repos).

Reply:
Would there be any value to be using the same fedora version across the board? (discussion independent of this PR)

Reply:
We can try but it would require always synchronizing changes in 3 repos (across 2 different orgs).  I'm not sure how realistic that is on the long term.

Reply:
Probably worth revisiting if we switch PR jobs to use ovn-org/ovn main